api/v1alpha1/proxmoxcluster_types.go

```diff
@@ -50,25 +48,52 @@ type ProxmoxClusterSpec struct {
 	// +optional
 	SchedulerHints *SchedulerHints `json:"schedulerHints,omitempty"`
 
+	ClusterNetworkConfig `json:",inline"`
+}
+
+// ClusterNetworkConfig represents information about the cluster network configuration.
+type ClusterNetworkConfig struct {
 	// IPv4Config contains information about available IPV4 address pools and the gateway.
-	// this can be combined with ipv6Config in order to enable dual stack.
+	// this can be combined with ipv6Config to enable dual stack.
 	// either IPv4Config or IPv6Config must be provided.
 	// +optional
-	// +kubebuilder:validation:XValidation:rule="self.addresses.size() > 0",message="IPv4Config addresses must be provided"
-	IPv4Config *ipamicv1.InClusterIPPoolSpec `json:"ipv4Config,omitempty"`
+	IPv4Config *IPConfig `json:"ipv4Config,omitempty"`
 
 	// IPv6Config contains information about available IPV6 address pools and the gateway.
-	// this can be combined with ipv4Config in order to enable dual stack.
+	// this can be combined with ipv4Config to enable dual stack.
 	// either IPv4Config or IPv6Config must be provided.
 	// +optional
-	// +kubebuilder:validation:XValidation:rule="self.addresses.size() > 0",message="IPv6Config addresses must be provided"
-	IPv6Config *ipamicv1.InClusterIPPoolSpec `json:"ipv6Config,omitempty"`
+	IPv6Config *IPConfig `json:"ipv6Config,omitempty"`
 
 	// DNSServers contains information about nameservers used by machines network-config.
 	// +kubebuilder:validation:MinItems=1
 	DNSServers []string `json:"dnsServers"`
 }
 
+// IPConfig contains information about available IP config.
+type IPConfig struct {
+	// Addresses is a list of IP addresses that can be assigned. This set of
+	// addresses can be non-contiguous.
+	// mutually exclusive with DHCP.
+	// +kubebuilder:validation:MinItems=1
+	Addresses []string `json:"addresses,omitempty"`
+
+	// Prefix is the network prefix to use.
+	// +kubebuilder:validation:Maximum=128
+	// +optional
+	Prefix int `json:"prefix,omitempty"`
+
+	// Gateway
+	// +optional
+	Gateway string `json:"gateway,omitempty"`
+
+	// DHCP indicates that DHCP should be used to assign IP addresses.
+	// mutually exclusive with Addresses.
+	// +optional
+	// +kubebuilder:default=false
```

Comment:
This default isn't needed. The behavior stays the same whether it's set to `false` or `null`. My pointer request was about making the user-facing API explicit and minimizing changes.